marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigBuilderImpl.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2012-2018 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.hub.impl;
+
+import com.marklogic.appdeployer.AppConfig;
+import com.marklogic.appdeployer.DefaultAppConfigFactory;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.HubConfigBuilder;
+import com.marklogic.mgmt.DefaultManageConfigFactory;
+import com.marklogic.mgmt.ManageClient;
+import com.marklogic.mgmt.ManageConfig;
+import com.marklogic.mgmt.admin.AdminConfig;
+import com.marklogic.mgmt.admin.AdminManager;
+import com.marklogic.mgmt.admin.DefaultAdminConfigFactory;
+import com.marklogic.mgmt.util.SimplePropertySource;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * A class for building a HubConfig class. Create a HubConfig instance like so:
+ *
+ * <pre>{@code
+ *     HubConfig hubConfig = HubConfigBuilder.newHubConfigBuilder("/path/to/your/project")
+ *         .withPropertiesFromEnvironment("local")
+ *         .build();
+ *}</pre>
+ */
+public class HubConfigBuilderImpl implements HubConfigBuilder {
+
+    private static final String GRADLE_PROPERTIES_FILENAME = "gradle.properties";
+
+    private HubConfigImpl hubConfig;
```

Comment:
Not sure about accepted best practice, but I'm inclined to use HubConfig here instead of HubConfigImpl, unless there's a compelling reason to do otherwise. Do we need to use the Impl class?

Reply:
I agree, unless there's some access to non-interface methods  in here.

Reply:
We do to encapsulate our reliance on the mgmt library we use to manage clients from being exposed through the interface.